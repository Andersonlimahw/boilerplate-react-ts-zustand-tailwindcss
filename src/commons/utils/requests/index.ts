export * from './api-config';
export * from './http-client';