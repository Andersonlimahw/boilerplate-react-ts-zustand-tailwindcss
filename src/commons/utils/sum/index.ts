interface SumInput {
  a: number;
  b: number;
}

export const sum = ( { a, b } : SumInput) => {
  return a + b;
}