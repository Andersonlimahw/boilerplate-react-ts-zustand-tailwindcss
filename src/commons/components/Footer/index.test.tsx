import { describe , expect, it } from 'vitest';

import { render, screen } from '../../utils/tests';
import { Footer } from '.';


describe('Footer', () => {
  it('render', () => {
    render(<Footer />);
    expect(screen.getByTestId('footer')).toBeInTheDocument();
  })

  it('snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  })

  it('render button_power', () => {
    const { getByTestId } = render(<Footer />);
    const powerButton = getByTestId('button_power');
    expect(powerButton).toBeInTheDocument();
  })
});