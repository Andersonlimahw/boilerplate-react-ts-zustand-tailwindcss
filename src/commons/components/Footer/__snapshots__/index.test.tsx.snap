// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-1/3 absolute bottom-10 flex flex-col"
    data-testid="footer"
  >
    <div
      class="py-2 px-3 bg-grey-lighter flex flex-row justify-between items-center"
    >
      <span
        class="cursor-pointer hover:scale-110 flex"
      >
        <svg
          class="mr-5"
          data-testid="button_power"
          fill="#ef4444"
          height="32"
          viewBox="0 0 256 256"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M120,128V48a8,8,0,0,1,16,0v80a8,8,0,0,1-16,0Zm60.37-78.7a8,8,0,0,0-8.74,13.4C194.74,77.77,208,101.57,208,128a80,80,0,0,1-160,0c0-26.43,13.26-50.23,36.37-65.3a8,8,0,0,0-8.74-13.4C47.9,67.38,32,96.06,32,128a96,96,0,0,0,192,0C224,96.06,208.1,67.38,180.37,49.3Z"
          />
        </svg>
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          aria-label="Theme switcher"
          class="ml-5 flex"
          data-state="closed"
          id="radix-:r2:"
          type="button"
        >
          <svg
            fill="#f1f1f1"
            height="32"
            viewBox="0 0 256 256"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M200.77,53.89A103.27,103.27,0,0,0,128,24h-1.07A104,104,0,0,0,24,128c0,43,26.58,79.06,69.36,94.17A32,32,0,0,0,136,192a16,16,0,0,1,16-16h46.21a31.81,31.81,0,0,0,31.2-24.88,104.43,104.43,0,0,0,2.59-24A103.28,103.28,0,0,0,200.77,53.89Zm13,93.71A15.89,15.89,0,0,1,198.21,160H152a32,32,0,0,0-32,32,16,16,0,0,1-21.31,15.07C62.49,194.3,40,164,40,128a88,88,0,0,1,87.09-88h.9a88.35,88.35,0,0,1,88,87.25A88.86,88.86,0,0,1,213.81,147.6ZM140,76a12,12,0,1,1-12-12A12,12,0,0,1,140,76ZM96,100A12,12,0,1,1,84,88,12,12,0,0,1,96,100Zm0,56a12,12,0,1,1-12-12A12,12,0,0,1,96,156Zm88-56a12,12,0,1,1-12-12A12,12,0,0,1,184,100Z"
            />
          </svg>
        </button>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
